src/plugins/discover/public/application/components/context_app/context_app.tsx

```diff
@@ -0,0 +1,177 @@
+      ...(fetchedState.anchor._id ? [fetchedState.anchor] : []),
+      ...(fetchedState.successors || []),
+    ],
+    [fetchedState.predecessors, fetchedState.anchor, fetchedState.successors]
+  );
+
+  const addFilter = useCallback(
+    async (field: IndexPatternField | string, values: unknown, operation: string) => {
+      const newFilters = esFilters.generateFilters(
+        filterManager,
+        field,
+        values,
+        operation,
+        indexPatternId
+      );
+      filterManager.addFilters(newFilters);
+      if (indexPatterns) {
+        const fieldName = typeof field === 'string' ? field : field.name;
+        await popularizeField(indexPattern, fieldName, indexPatterns);
+      }
+    },
+    [filterManager, indexPatternId, indexPatterns, indexPattern]
+  );
+
+  const TopNavMenu = navigation.ui.TopNavMenu;
+  const getNavBarProps = () => {
+    return {
+      appName: 'context',
+      showSearchBar: true,
+      showQueryBar: false,
+      showFilterBar: true,
+      showSaveQuery: false,
+      showDatePicker: false,
+      indexPatterns: [indexPattern],
+      useDefaultBehaviors: true,
+    };
+  };
+
+  return (
+    <I18nProvider>
+      {fetchedState.anchorStatus.value === LoadingStatus.FAILED ? (
+        <ContextErrorMessage status={fetchedState.anchorStatus} />
+      ) : (
+        <Fragment>
+          <TopNavMenu {...getNavBarProps()} />
+          <EuiPage className={classNames({ dscDocsPage: !isLegacy })}>
+            <EuiPageContent paddingSize="s" className="dscDocsContent">
+              <EuiSpacer size="s" />
+              <EuiText>
+                <strong>
+                  <FormattedMessage
+                    id="discover.context.contextOfTitle"
+                    defaultMessage="Documents surrounding #{anchorId}"
+                    values={{ anchorId }}
+                  />
+                </strong>
+              </EuiText>
+              <EuiSpacer size="s" />
+              <ContextAppContentMemoized
+                columns={columns}
+                onAddColumn={onAddColumn}
+                onRemoveColumn={onRemoveColumn}
+                onSetColumns={onSetColumns}
+                services={services}
+                indexPattern={indexPattern}
+                predecessorCount={appState.predecessorCount}
+                successorCount={appState.successorCount}
+                rows={rows}
+                sort={appState.sort as [[string, SortDirection]]}
+                predecessors={fetchedState.predecessors}
+                successors={fetchedState.successors}
+                anchorStatus={fetchedState.anchorStatus.value}
+                predecessorsStatus={fetchedState.predecessorsStatus.value}
+                successorsStatus={fetchedState.successorsStatus.value}
+                useNewFieldsApi={useNewFieldsApi}
+                isLegacy={isLegacy}
+                setAppState={setAppState}
+                addFilter={addFilter}
```

Comment:
just a personal preference, the longer such property list are the harder to read. Personally I use a "Sort lines" command in Intellij, to sort A-Z.  On the other side, then there's no grouping by functionality like columns. However, just loud thinking, it fine if it stays like this.